wsmaster/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/jpa/JpaSnapshotDao.java

```diff
@@ -161,4 +172,10 @@ protected void doRemove(String snapshotId) throws NotFoundException {
         newSnapshots.forEach(manager::persist);
         return existing;
     }
+
+    @Transactional
+    protected SnapshotImpl doUpdate(SnapshotImpl update) {
+        return managerProvider.get().merge(update);
```

Comment:
1. `update` may not be managed by jpa
2. if `update` is managed, it may be either inserted or updated, consider using get + merge
3. Usually update methods throw `NotFoundException` when there is no entity with given identifier